### PR TITLE
Require driver class in Iceberg JDBC catalog 

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Require driver class in Iceberg JDBC catalog. 

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Iceberg
* Require `iceberg.jdbc-catalog.driver-class` config property to prevent driver not found error after the initialization ({issue}`16196`)
```

### CODE REVIEW

**Comment by ebyhr** [comment]
`web-ui-checks` hit #16179

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/IcebergJdbcCatalogConfig.java [hunk 1]
```diff
@@ -24,12 +24,26 @@
...
     private String defaultWarehouseDir;
 
+    @NotNull
+    public String getDriverClass()
+    {
+        return driverClass;
+    }
+
+    @Config("iceberg.jdbc-catalog.driver-class")
+    public IcebergJdbcCatalogConfig setDriverClass(String driverClass)
```
Comment (krvikash) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/IcebergJdbcCatalogConfig.java:41]: Can we have `@ConfigDescription` as well?

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/IcebergJdbcConnectionFactory.java [hunk 2]
```diff
@@ -16,32 +16,38 @@
...
 import static java.util.Objects.requireNonNull;
 
 public class IcebergJdbcConnectionFactory
         implements ConnectionFactory
 {
+    private final Driver driver;
     private final String connectionUrl;
-    private final Optional<String> user;
-    private final Optional<String> password;
+    private final Optional<String> connectionUser;
```
Comment (krvikash) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/IcebergJdbcConnectionFactory.java:32]: IMO renaming of variables should be in a separate commit.

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 3]
```diff
@@ -141,6 +141,7 @@ When using any database besides PostgreSQL, a JDBC driver jar file must be place
     connector.name=iceberg
     iceberg.catalog.type=jdbc
     iceberg.jdbc-catalog.catalog-name=test
+    iceberg.jdbc-catalog.driver-class=org.postgresql.Driver
```
Comment (krvikash) [inline docs/src/main/sphinx/connector/iceberg.rst:144]: Do we need to include driver-class property in the minimum requirenments?

```
At a minimum, ``iceberg.jdbc-catalog.connection-url`` and
``iceberg.jdbc-catalog.catalog-name`` must be configured.
```
https://github.com/trinodb/trino/pull/16196/commits/4400fc0ef7e044187986bdf15c66f7005bd07ed7#diff-e1aabf1cfd8bd8aa7d1b75e70089b57413b2e620a5eebeb36cc76fd3f2ac60dbR135-R136

**Review by krvikash — COMMENTED** [review]
LGTM. a few comments.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/IcebergJdbcCatalogConfig.java [hunk 4]
```diff
@@ -24,12 +24,27 @@
 
 public class IcebergJdbcCatalogConfig
 {
+    private String driverClass;
```
Comment (losipiuk) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/IcebergJdbcCatalogConfig.java:27]: nit: maybe allow not setting the driver if url is using `jdbc:postgresql` schema

**Review by losipiuk — APPROVED** [review]

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/IcebergJdbcCatalogConfig.java (hunk 4 above)
Comment (ebyhr) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/IcebergJdbcCatalogConfig.java:27]: It may cause a minor issue if third-party JDBC developers re-use the prefix. I suppose CData uses the prefix with the different driver class `cdata.jdbc.postgresql.PostgreSQLDriver`. Let me keep as-is.
